internal/controller/services/gateway/gateway_controller_actions.go

```diff
@@ -58,52 +59,246 @@ func createGatewayInfrastructure(ctx context.Context, rr *odhtypes.Reconciliatio
 		return fmt.Errorf("failed to create GatewayClass: %w", err)
 	}
 
-	certSecretName, err := handleCertificates(ctx, rr, gatewayConfig, domain)
+	certSecretName, err := handleCertificates(ctx, rr, gatewayConfig, hostname)
 	if err != nil {
 		return fmt.Errorf("failed to handle certificates: %w", err)
 	}
 
-	if err := createGateway(rr, certSecretName, domain, DefaultGatewayName); err != nil {
+	if err := createGateway(rr, certSecretName, hostname); err != nil {
 		return fmt.Errorf("failed to create Gateway: %w", err)
 	}
 
 	l.V(1).Info("Successfully created Gateway infrastructure",
 		"gateway", DefaultGatewayName,
 		"namespace", GatewayNamespace,
-		"domain", domain,
+		"domain", hostname,
 		"certificateType", getCertificateType(gatewayConfig))
 
 	return nil
 }
 
-// createDestinationRule creates a DestinationRule for TLS configuration using embedded YAML template.
-// This function uses embedded resources for efficient template management.
-func createDestinationRule(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	// Input validation
-	if rr == nil {
-		return errors.New("reconciliation request cannot be nil")
+// Check authentication mode and deploy auth proxy (secret + service + deployment) + OAuth client (if integrated mode) + HTTPRoute + DestinationRule.
+func createKubeAuthProxyInfrastructure(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	l := logf.FromContext(ctx).WithName("createAuthProxy")
+
+	gatewayConfig, err := validateGatewayConfig(rr)
+	if err != nil {
+		return err
 	}
 
-	l := logf.FromContext(ctx).WithName("createDestinationRule")
-	l.V(1).Info("Creating DestinationRule for TLS configuration")
+	l.V(1).Info("creating auth proxy for gateway", "gateway", gatewayConfig.Name)
 
-	yamlContent, err := gatewayResources.ReadFile(destinationRuleTemplate)
+	authMode, err := cluster.GetClusterAuthenticationMode(ctx, rr.Client)
 	if err != nil {
-		return fmt.Errorf("failed to read DestinationRule template: %w", err)
+		return fmt.Errorf("failed to detect cluster authentication mode: %w", err)
+	}
+	l.V(1).Info("detected cluster authentication mode", "mode", authMode)
+
+	// Reset the condition to false
+	rr.Conditions.MarkFalse(ReadyConditionType)
+
+	kubeAuthProxyDeploymentTemplates := odhtypes.TemplateInfo{
+		FS:   gatewayResources,
+		Path: kubeAuthProxyDeploymentOauthTemplate,
 	}
 
-	decoder := serializer.NewCodecFactory(rr.Client.Scheme()).UniversalDeserializer()
-	unstructuredObjects, err := resources.Decode(decoder, yamlContent)
+	var oidcConfig *serviceApi.OIDCConfig
+	switch authMode {
+	case cluster.AuthModeOIDC:
+		if gatewayConfig.Spec.OIDC == nil {
+			rr.Conditions.MarkFalse(
+				ReadyConditionType,
+				conditions.WithReason(status.NotReadyReason),
+				conditions.WithMessage(status.AuthProxyOIDCModeWithoutConfigMessage),
+			)
+			return nil // TODO: is this logic correct? no oidc but to use oidc and not error out.
```

Comment:
```suggestion
          // Return nil (not error) because this is a permanent user configuration error.
          // Returning an error would cause infinite reconciliation for something only the
          // user can fix. Users must check GatewayConfig status conditions to see this error.
          // Reconciliation will retry when the user updates the GatewayConfig with OIDC config.
          return nil
```

We have the same pattern in the monitoring controller where it does the same thing when metrics/traces aren't configured.

However, we should make this even better in future PRs:
- Admission webhook: It would be nice to validate the OIDC config at admission time so users get immediate feedback instead of having to check status conditions. 
- Watch auth mode changes: Right now, if someone switches the cluster from OAuth to OIDC, we won't know until something else triggers reconciliation. Could watch the Authentication CR if you want to be more responsive.
- Alerting: Maybe add a PrometheusRule if the gateway sits in NotReady for too long?

@GowthamShanmugam, @zdtsw WDYT? We can create follow-up Jiras.